espei/utils.py

```diff
@@ -46,19 +46,47 @@ def __setstate__(self, state):
         self.__init__(storage=MemoryStorage)
         self.insert_multiple(state['_tables']['_default'])
 
+class PredictWrapper:
+    def __init__(self, client, f, **kwargs):
+        self.f = f
+        self.client = client
+
+        # TODO: I'm debating whether to submit a self generating lambda function for the context or to scatter the context
+        # as data. With scattering, the data will transfer before the worker starts anything, but I don't know
+        # if a worker that restarts will retain the data. The lambda function will transfer data after the worker starts
+        # (it should only be once at the start), but I think if a worker dies, it could still generate the data again (although,
+        # since the self generating function lives on a specific worker, I don't know what would happen if that specific worker
+        # restarts. Memory and performance-wise, scattering vs submitting lambda function seems to be the same
+        self.kwargs = {key: self.client.submit(lambda x: x, val, key=key) for key,val in kwargs.items()}
+        #self.kwargs = {key: self.client.scatter(val, broadcast=True, hash=False) for key,val in kwargs.items()}
```

Comment:
Update on this TODO? Maybe it can be tested in a hacky way by modifying the main optimizer loop to restart the workers every iteration and observe whether everything still works and if there's memory growth one way or the other.